bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerChecker.java

```diff
@@ -195,10 +199,45 @@ private void verifyLedgerFragment(LedgerFragment fragment,
             bookieClient.readEntry(bookie, fragment
                     .getLedgerId(), firstStored, manycb, null);
         } else {
-            ReadManyEntriesCallback manycb = new ReadManyEntriesCallback(2,
+            assert lastStored > firstStored;
+            long lengthOfLedgerFragment = lastStored - firstStored + 1;
+
+            int numberOfEntriesToBeVerified =
+                (int) (lengthOfLedgerFragment * (percentageOfLedgerFragmentToBeVerified / 100.0));
+
+            TreeSet<Long> entriesToBeVerified = new TreeSet<Long>();
+
+            if (numberOfEntriesToBeVerified < lengthOfLedgerFragment) {
+                // Evenly pick random entries over the length of the fragment
+                if (numberOfEntriesToBeVerified != 0) {
+                    int lengthOfBucket = (int) (lengthOfLedgerFragment / numberOfEntriesToBeVerified);
+                    Random rand = new Random();
```

Comment:
use [ThreadLocalRandom.current()](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadLocalRandom.html#current()) rather than constructing a new `Random` object.

Reply:
will change it